product-service/src/functions/products/addProduct.ts

```diff
@@ -0,0 +1,32 @@
+import { APIGatewayProxyResult } from 'aws-lambda';
+
+import { formatJSONResponse } from '../../libs/api-gateway';
+import { middyfy } from '../../libs/lambda';
+import { Product } from '../../models/fe/product.model';
+import { insertProduct } from '../../utils/product/insert';
+import { clientError, serverError } from '../../utils/errors';
+import { log } from '../../utils/logger';
+
+export const addProduct = async (event: { body: Omit<Product, 'id'> }): Promise<APIGatewayProxyResult> => {
+  try {
+    log('addProduct', event);
+    
+    const product: Product = await insertProduct(event.body);
+    if (!product) {
+      const { statusCode, message } = clientError;
+      return formatJSONResponse(statusCode, {
+        message
+      });
+    }
+    return formatJSONResponse(201, {
+      product,
+    });
```

Comment:
Also here you don't need to wrap with the object with `product` field, you can just pass `product` to get rid of these wrappers.

```suggestion
    return formatJSONResponse(201, product)
```